src/core/trulens/core/utils/python.py

```diff
@@ -1148,118 +1150,31 @@ def rewrap(obj_):
 
 
 # Class utilities
-
-
-@dataclasses.dataclass
-class SingletonInfo(Generic[T]):
-    """
-    Information about a singleton instance.
+class SingletonPerName(type):
```

Comment:
Change name so it is obvious that it needs to be set as metaclass instead of parent. Perhaps `MetaSingletonPerName` ?